asm/callonstack.s

```diff
@@ -8,13 +8,25 @@
 //     [RSI] long (*func)(void* arg),
 //     [RDX] void* arg);
 //
-// Call a function on the given stack
+// Call a function on the given stack.
 //
+// GDB normally will stop stack-walking at a function if it determines that
+// the stack does not monotonically decrease at the function.
+// This could be the case for some invocations of myst_call_on_stack, if the
+// `stack` parameter is greater than RSP upon entry.
+// To support the split-stack feature, GDB however relaxes the above
+// constraint for functions named `__morestack`.
```

Comment:
This is interesting. We have a few other places in our code where we walk stacks... getting backtrace for one and our pseudo-fork implementation where we copy stack and fix-up stack frames. Wondering if we can leverage this fact in these other places to make it more robust.

Reply:
Can you point me to the places in code you are referring to and where to put a breakpoint to see how GDBs backtrace currently appears ?